internal/xdscache/v3/route.go

```diff
@@ -303,12 +318,12 @@ func (v *routeVisitor) visit(vertex dag.Vertex) {
 }
 
 // sortRoutes sorts the given Route slice in place. Routes are ordered
-// first by longest prefix (or regex), then by the length of the
-// HeaderMatch slice (if any). The HeaderMatch slice is also ordered
-// by the matching header name.
-func sortRoutes(routes []*envoy_route_v3.Route) {
+// first by path match type, path match value via string comparison and
+// then by the length of the HeaderMatch slice (if any). The HeaderMatch
+// slice is also ordered by the matching header name.
+func sortRoutes(routes []*dag.Route) {
```

Comment:
not sure where the best place is, but it'd be good to add a comment about *why* we sort the DAG routes here vs. the Envoy ones. That may make more sense with the subsequent PR, though.

Reply:
Added a comment, LMK what you think!

Reply:
:+1: I think that's helpful